internal/services/resource/resource_group_resource.go

```diff
@@ -209,3 +251,53 @@ https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#features
 `, name, strings.Join(formattedResourceUris, "\n"))
 	return fmt.Errorf(strings.ReplaceAll(message, "'", "`"))
 }
+
+func resourceExistById(ctx context.Context, resourceClient *resources.Client, resourceId string, apiVersion string) (exist bool, err error) {
+	resource, err := resourceClient.GetByID(ctx, resourceId, apiVersion)
+	if err != nil {
+		if utils.ResponseWasNotFound(resource.Response) {
+			return false, nil
+		}
+		return true, err
+	}
+
+	return resource.ID != nil, nil
+}
+
+func getResourceTypeApiVersion(ctx context.Context, client *resources.ProvidersClient, resource resources.GenericResourceExpanded) (apiVersion string, err error) {
+	providerName, targetResourceType, err := splitProviderAndResourceTypes(*resource.Type)
+	if err != nil {
+		return "", err
+	}
+
+	provider, err := client.Get(ctx, providerName, fmt.Sprintf("resourceTypes/%s", targetResourceType))
+	if err != nil {
+		return "", err
+	}
+
+	if provider.ResourceTypes == nil {
+		return "", fmt.Errorf("getting resource(%s/%s) default api version, the provider returned nil on resource type. ", providerName, targetResourceType)
+	}
+
+	for _, resourceType := range *provider.ResourceTypes {
+		if resourceType.ResourceType != nil && *resourceType.ResourceType == targetResourceType {
+			if resourceType.DefaultAPIVersion != nil && *resourceType.DefaultAPIVersion != "" {
+				return *resourceType.DefaultAPIVersion, nil
+			} else if len(*resourceType.APIVersions) > 0 && (*resourceType.APIVersions)[0] != "" {
+				return (*resourceType.APIVersions)[0], nil
+			}
+
+			return "", fmt.Errorf("getting resource(%s/%s) api version, there is no available api version. ", providerName, targetResourceType)
+		}
+	}
+
+	return "", fmt.Errorf("getting resource(%s/%s) api version, there is no match resource type. ", providerName, targetResourceType)
+}
+
+func splitProviderAndResourceTypes(fullResourceType string) (provider string, resourceType string, err error) {
+	p, r, found := strings.Cut(fullResourceType, "/")
```

Comment:
this assumes a 1:1 between RP and Resource Type when they can be both `Microsoft.Compute/virtualMachines` and `Microsoft.Compute/virtualMachines/extensions`?

Reply:
It takes the string in front of the first "/" as the RP and the rest part is Resource Type.

Reply:
we assusme RP always before the first slash